repository.go

```diff
@@ -453,3 +454,129 @@ func (r *Repository) Reference(name plumbing.ReferenceName, resolved bool) (
 func (r *Repository) References() (storer.ReferenceIter, error) {
 	return r.s.IterReferences()
 }
+
+// ResolveRevision resolves revision to corresponding hash
+func (r *Repository) ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, error) {
+	p := revision.NewParserFromString(string(rev))
```

Comment:
I'm sorry, I don't understand @mcuadros comment either.

Maybe he has something in mind about how this is going to be used in the future.

Apparently he wants the `ResolveRevision` method from `Repository` to receive a plain string, not a `plumbing.Revision`. I'm not sure if he wants the method to return also strings instead of a `plumbing.Hash`.

Let's invoke him, so he can solve the problem: @mcuadros, please check this!. You should do what he says if you want your PR to be merged.

PS: independently of that, if a revision will represents revisons ranges in the future, this method should return a slice of strings or of hashes, just to take that into account future improvements.